src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,241 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+import java.util.Scanner;
+
+public class Duke {
+    private static final String LS = System.lineSeparator();
+
+    /* Predetermined Message List */
+    private static final String MESSAGE_LOGO = " ____        _        "+LS
+            + "|  _ \\ _   _| | _____ "+LS
+            + "| | | | | | | |/ / _ \\"+LS
+            + "| |_| | |_| |   <  __/"+LS
+            + "|____/ \\__,_|_|\\_\\___|"+LS;
+    private static final String MESSAGE_INTRO = "\t Hello! I'm Duke"
+            + LS +"\t What can I do for you?";
+    private static final String MESSAGE_EXIT = "\t Bye. Hope to see you again soon!";
+    private static final String MESSAGE_DONE = "\t Nice! I've marked this task as done:";
+    private static final String MESSAGE_TASK_ADDED = "\t Got it. I've added this task:";
+    private static final String MESSAGE_LIST = "\t Here are the tasks in your list:";
+    private static final String MESSAGE_LINE = "\t__________________________________________________________________________________________";
+    private static final String MESSAGE_INVALID_COMMAND_ERROR = "\t ☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+
+    /* Command List */
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_EXIT = "bye";
+
+    /* Command Separator Parameter List */
+    private static final String PARAM_DELIMIT_BY = " /by ";
+    private static final String PARAM_DELIMIT_AT = " /at ";
+    private static final int PARAM_DELIMIT_LIMIT = 2;
+
+    private static Task[] tasks = new Task[100];
```

Comment:
I like how all the constants are named and used in the code. Really improved the code's readability.